kotest-framework/kotest-framework-multiplatform-plugin-gradle/src/main/kotlin/io/kotest/framework/multiplatform/gradle/KotestMultiplatformCompilerGradlePlugin.kt

```diff
@@ -38,40 +39,51 @@ class KotestMultiplatformCompilerGradlePlugin : KotlinCompilerPluginSupportPlugi
     * Takes the version from the gradle extension configuration first, or if not
     * specified, then defaults to using the same version as the engine dependency.
     */
-   private fun version(): String? {
-      val project = target ?: error(missingProjectValError)
-
+   private val version: String? by lazy {
       val versionFromExtension = extension?.compilerPluginVersion?.orNull
       if (versionFromExtension != null) {
          println("Kotest compiler plugin [$versionFromExtension]")
-         return versionFromExtension
+         return@lazy versionFromExtension
       }
 
-      val version = engineDeps(project).firstOrNull()?.version ?: return null
+      val engineDep = engineDeps().firstOrNull() ?: error("Cannot determine Kotest compiler plugin version if no Kotest engine dependencies are present")
+      val version = engineDep.version ?: return@lazy null
+
       if (version.contains("LOCAL")) {
          println("Detected dev engine version [$version]")
       }
-      return version
+
+      return@lazy version
    }
 
-   private fun engineDeps(project: Project) =
-      project.configurations
+   private fun engineDeps(): List<Dependency> {
+      val project = target ?: error(missingProjectValError)
+
+      return project.configurations
          .flatMap { it.all }
          .flatMap { it.dependencies }
          .filter { it.group == KotestGroupId && it.name.startsWith(engineDepPrefix) }
+   }
 
    override fun getCompilerPluginId() = compilerPluginId
 
    override fun getPluginArtifact(): SubpluginArtifact =
-      SubpluginArtifact(KotestGroupId, KotestJsArtifactId, version())
+      SubpluginArtifact(KotestGroupId, KotestJsArtifactId, version)
 
    override fun getPluginArtifactForNative(): SubpluginArtifact =
-      SubpluginArtifact(KotestGroupId, KotestNativeArtifactId, version())
+      SubpluginArtifact(KotestGroupId, KotestNativeArtifactId, version)
 
    override fun isApplicable(kotlinCompilation: KotlinCompilation<*>): Boolean {
       return when {
          // if we can't find a version to use then we won't apply to this module
-         version() == null -> false
+         engineDeps().isEmpty() -> {
```

Comment:
If this is empty, would the error on 49 be triggered ?

Reply:
The error on line 49 won't be triggered when `engineDeps()` is called. I added this check to maintain the existing behaviour (not enabling the plugin and continuing without error if no Kotest dependencies are found). 

Personally, I'd prefer it if the plugin errored if it can't find a valid dependency, as that likely isn't what the user intended.